### PR TITLE
Relax detached triplet filter

### PR DESCRIPTION
**Pull request by mmasciov**
### PR description

Updated (WP of the) detachedTripletStep quality filter is aimed at improving "Built track" efficiency as much as possible, with ~ no increase in fake rate. No change is expected in other iterations.

It requires update of mkFit JSON files: trackreco/cmssw#55

### PR validation (PR #365 + trackreco/cmssw#55):

Expectation described above is confirmed by MTV results collected below.
--> PRs are ready to be merged (in absence of objections).

(1) NO change in mkFit-4iter: 
http://uaf-10.t2.ucsd.edu/~mmasciov/MkFit_devs/SVA_displacedIters/detachedTripletFilter/MTV_mkFit-4iter_PR365/

(2) Slight increase in efficiency up to "All tracks" for mkFit-7iter, due to increase in efficiency for detachedTripletStep: http://uaf-10.t2.ucsd.edu/~mmasciov/MkFit_devs/SVA_displacedIters/detachedTripletFilter/MTV_mkFit-7iter_PR365_all/

(3) When mkFit is enabled only in detachedTripletStep, improvement in efficiency up to "All tracks": 
http://uaf-10.t2.ucsd.edu/~mmasciov/MkFit_devs/SVA_displacedIters/detachedTripletFilter/MTV_mkFit-detachedTripletStep_PR365_all/

(4) When mkFit is enabled in 4-iter + detachedTripletStep, improvement in efficiency up to "All tracks":
http://uaf-10.t2.ucsd.edu/~mmasciov/MkFit_devs/SVA_displacedIters/detachedTripletFilter/MTV_mkFit-4iter-plusDetachedTripletStep_PR365_all/

### CODE REVIEW

File: mkFit/MkStdSeqs.h [hunk 1]
```diff
@@ -51,8 +48,26 @@ namespace StdSeq
...
+      float pt_min = 0.7; // min pT for full filter
+      float d0BS   = t.d0BeamSpot(bspot.x,bspot.y);
+      float d0_max = 0.1; // 1 mm
+      
+//      return !((nhits ==3 && (llyr==2||llyr==18||llyr==45)) ||
+//	       (layers==3 && (llyr==2||llyr==18||llyr==45)));
+      return !((nhits ==3 && (llyr==2||llyr==18||llyr==45) && pt> pt_min) ||
+	       (layers==3 && (llyr==2||llyr==18||llyr==45) && pt> pt_min) ||
+	       (layers==3 && (llyr==2||llyr==18||llyr==45) && pt<=pt_min && d0BS>d0_max) ||
+	       (nhits ==3 && (llyr==2||llyr==18||llyr==45) && pt<=pt_min && d0BS>d0_max));
```
Comment (slava77) [inline mkFit/MkStdSeqs.h]:
```suggestion
      bool endsInsidePix = (llyr==2||llyr==18||llyr==45);
      return !( (nhits ==3 || layers==3) && endsInsidePix && (pt> pt_min || (pt<=pt_min && d0BS>d0_max) ) );
```
seems shorter

File: mkFit/buildtestMPlex.cc [hunk 2]
```diff
@@ -512,7 +512,7 @@ std::vector<double> runBtpCe_MultiIter(Event& ev, const EventOfHits &eoh, MkBuil
       builder.filter_comb_cands([&](const TrackCand &t)
         { return StdSeq::qfilter_n_hits(t, itconf.m_params.minHitsQF); });
     }
-    else if (itconf.m_track_algorithm==6)
+    if (itconf.m_track_algorithm==6)
```
Comment (slava77) [inline mkFit/buildtestMPlex.cc]: as mentioned on skype, need to at least add a FIXME: to have a consistent `m_require_quality_filter` in configuration

File: mkFit/MkStdSeqs.h (hunk 1 above)
Comment (mmasciov) [reply mkFit/MkStdSeqs.h]: Please, check updated version. Thanks!

File: mkFit/buildtestMPlex.cc (hunk 2 above)
Comment (mmasciov) [reply mkFit/buildtestMPlex.cc]: Please, check updated version. Thanks!

**Comment by mmasciov** [comment]
### PR validation (PR #365 + trackreco/cmssw#55):

Updated (WP of the) detachedTripletStep quality filter is aimed at improving "Built track" efficiency as much as possible, with ~ no increase in fake rate. No change is expected in other iterations.
This is confirmed by MTV results collected below.
--> PRs are ready to be merged (in absence of objections).

(1) NO change in mkFit-4iter: 
http://uaf-10.t2.ucsd.edu/~mmasciov/MkFit_devs/SVA_displacedIters/detachedTripletFilter/MTV_mkFit-4iter_PR365/

(2) Slight increase in efficiency up to "All tracks" for mkFit-7iter, due to increase in efficiency for detachedTripletStep: http://uaf-10.t2.ucsd.edu/~mmasciov/MkFit_devs/SVA_displacedIters/detachedTripletFilter/MTV_mkFit-7iter_PR365_all/

(3) When mkFit is enabled only in detachedTripletStep, improvement in efficiency up to "All tracks": 
http://uaf-10.t2.ucsd.edu/~mmasciov/MkFit_devs/SVA_displacedIters/detachedTripletFilter/MTV_mkFit-detachedTripletStep_PR365_all/

(4) When mkFit is enabled in 4-iter + detachedTripletStep, improvement in efficiency up to "All tracks":
http://uaf-10.t2.ucsd.edu/~mmasciov/MkFit_devs/SVA_displacedIters/detachedTripletFilter/MTV_mkFit-4iter-plusDetachedTripletStep_PR365_all/

**Pull request merged**